src/plugins/google-analytics/server.js

```diff
@@ -0,0 +1,76 @@
+var google = require('googleapis');
+var _ = require('lodash');
+var util = require('util');
+
+var analytics = google.analytics('v3');
+var validator = require('./validator');
+
+var exports = module.exports = {};
+var authClient;
+
+function initDatasource(datasource, io) {
+
+  var key = require('../../../config/' + datasource.keyFile);
+  authClient = new google.auth.JWT(
+    key.client_email,
+    null,
+    key.private_key,
+    ['https://www.googleapis.com/auth/analytics.readonly'],
+    null
+  );
+
+  refreshDatasource(datasource, io);
+  setInterval(function() {
+    refreshDatasource(datasource, io);
+  }, datasource.updateInterval);
+}
+
+function gaRequest(viewId, metrics, filters, dimensions) {
+  return new Promise(function (resolve, reject) {
+    authClient.authorize(function(err, tokens) {
+      if (err) {
+        reject(Error());
+        return;
+      }
+      var getParams = {
+        auth: authClient,
+        'ids': viewId,
+        'metrics': metrics
+      };
+      if (filters !== undefined) {
+        getParams.filters = filters;
+      }
+      if (dimensions !== undefined) {
+        getParams.dimensions = dimensions;
+      }
```

Comment:
Could this part be simplified to just 
```
var getParams = {
          auth: authClient,
          'ids': viewId,
          metrics,
          filters,
          dimensions
        };
```
Or does the realtime client throw? (need to set `"es6":true` in eslintrc environments for shorthand initializers)